CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -386,7 +386,7 @@ function Staff:checkIfNeedRest()
       self:changeAttribute("happiness", -0.001)
     end
     local room = self:getRoom()
-    if (self.staffroom_needed and ((room and not room:getPatient()) or not room)) or
+    if (self.staffroom_needed and (room and (not room:getPatient() or room.room_info.id == "ward")) or not room) or
         (room and self.going_to_staffroom) then
```

Comment:
An if-else with more than one 'and' and one 'or' is a sign is it too complicated.

No doubt it is correct, but it gives me a headache already without even trying to understand what it says.

Please refactor it into a bunch of simple understandable 'if' statements, of make a function that returns a boolean result for this computation.